### DISCUSSION TITLE
How to redeploy when adding not-nullable field to datamodel.graphql?

### DISCUSSION THREAD

**Discussion by amin-wan-kenobi** [prisma/prisma#2190] [Q&A] [answered]
I have created my datamodel.graphql and I added some data to the database to test. Now I added a not-nullable field to the type in my datamodel.graphql but I cannot deploy my prisma because it does not allow it due to having not-nullable fields. Of course I can run

prisma delete  and redeploy it again but let's say we are in the production and deleting is not an option. How can we handle this?

**Comment by janpio** [comment, marked as answer]
This sounds like a Prisma 1 question. Prisma 1's repository was moved to https://github.com/prisma/prisma1, so you might want to ask your question there again to get an answer.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
